test/nfs.js

```diff
@@ -200,12 +209,63 @@ describe('NFS emulation', () => {
     const size = await file.size();
     return should(size).be.equal(18);
   });
-//  describe('internal function', () => {
-//    it.only('verifies if file meta data is String instance', async () => {
-//      const mData = await app.mutableData.newRandomPublic(TYPE_TAG);
-//      await mData.quickSetup({});
-//      const nfs = await mData.emulateAs('nfs');
-//      const file = await nfs.open(null, CONSTANTS.NFS_FILE_MODE_OVERWRITE);
-//    });
-//  });
+
+  it('creates file with user metadata', async () => {
+    const m = await app.mutableData.newRandomPublic(TYPE_TAG);
+    await m.quickSetup({});
+    const nfs = await m.emulateAs('nfs');
+    const userMetadata = 'text/plain';
+    const file = await nfs.create('hello, SAFE world!', userMetadata);
+    return should(file.userMetadata.toString())
+      .be.equal(userMetadata);
+  });
+
+  it('fetches file and reads user meta data', async () => {
+    const m = await app.mutableData.newRandomPublic(TYPE_TAG);
+    await m.quickSetup({});
+    const nfs = await m.emulateAs('nfs');
+    const userMetadata = 'text/plain';
+    let file = await nfs.create('hello, SAFE world!', userMetadata);
+    await nfs.insert('hello.txt', file);
+    file = await nfs.fetch('hello.txt');
+    return should(file.userMetadata.toString())
+      .be.equal(userMetadata);
+  });
+
+  it('updates file user metadata', async () => {
+    const m = await app.mutableData.newRandomPublic(TYPE_TAG);
+    await m.quickSetup({});
+    const nfs = await m.emulateAs('nfs');
+    const userMetadata = 'text/plain';
+    let file = await nfs.create('hello, SAFE world!', userMetadata);
+    should(file.userMetadata.toString()).be.equal(userMetadata);
+    await nfs.insert('hello.txt', file);
+    file = await nfs.fetch('hello.txt');
+    const fileVersion = file.version;
+    await nfs.update('hello.txt', file, fileVersion + 1, 'text/javascript');
+    file = await nfs.fetch('hello.txt');
+    return should(file.userMetadata.toString())
+      .be.equal('text/javascript');
+  });
+
+  it('throws error if invalid user metadata type is passed while updating file', async () => {
+    const m = await app.mutableData.newRandomPublic(TYPE_TAG);
+    await m.quickSetup({});
+    const nfs = await m.emulateAs('nfs');
+    let file = await nfs.create('hello, SAFE world!', 'text/plain');
+    await nfs.insert('hello.txt', file);
+    should(file.ref.user_metadata_len).be.equal(10);
```

Comment:
We shouldn't do redundant steps in our tests, if this same verification (length of the metadata in this case) is made in previous tests we can skip it here, and go straight to the other type of verification we are aiming at, otherwise we have a set of tests which are bigger and bigger but don't actually provide more coverage.